### DISCUSSION TITLE
Supporting all software?

### DISCUSSION THREAD

**Discussion by joachimlindenberg** [ChrisTitusTech/winutil#1361] [Q&A]
I´d really love to get rid of some of my scripts and tweaks! The statement "[Upgrade all existing programs](https://github.com/ChrisTitusTech/winutil#:~:text=Upgrade%20All:%20Upgrade%20all%20existing%20programs)" is very bold. How do you detect installed software (w/o portable software) and how do you determine upgrades available, or do you report on software installed but not supported?

**Comment by REVENGE977** [comment]
There is a command in [winget](https://github.com/microsoft/winget-cli) that updates all of the installed programs. His tool just runs that command.

**Comment by joachimlindenberg** [comment]
winget updates packages from some repositories, but afaik most windows software is distributed through vendor downloads. Afaik winget does not claim to update all installed programs. Can you please reference a quote? Or consider revising the statement?

**Reply by REVENGE977** [reply]
Well, not **every single app**, but most apps are on winget. The command will upgrade all installed apps, as long as they do exist on winget, even if an app wasn't installed through winget by the user.

**Comment by joachimlindenberg** [comment]
I checked some of the apps I have installed on winget.run. Many are just missing, most available are outdated. Outdated is probably even worse than missing from a security point of view.

**Reply by REVENGE977** [reply]
I believe if the version you have is higher it won't "upgrade."

**Comment by joachimlindenberg** [comment]
I assume that winget.run is maintained by volunteers and not software vendors. Unless the vendor of the software commits to providing updates via winget.run, one cannot and should not depend on winget, nor claim it updates all software (which winget doesn´t, but this tool does).
